include/seqan3/core/metafunction/basic.hpp

```diff
@@ -117,8 +117,60 @@ struct remove_rvalue_reference
 template <typename t>
 using remove_rvalue_reference_t = typename remove_rvalue_reference<t>::type;
 
-//!\}
+// ----------------------------------------------------------------------------
+// deferred_type
+// ----------------------------------------------------------------------------
+
+/*!\brief Return the type identity; further arguments are ignored, but can make this type dependent if they are
+ *        [TransformationTrait].
+ * \tparam t The type to operate on.
+ * \tparam dependent_ts Any provided types are ignored.
+ */
+template <typename t, typename ...dependent_ts>
+struct deferred_type
```

Comment:
I don't think that this should be public API. The use case is very special and would require at least a snippet to get the idea. Instead I would put it into detail and maybe even in the adl stuff/customisation point stuff.

Reply:
I can think of other places where this could be useful, but I can also move it to detail, if you prefer.